internal/resources/azure/data_factory_integration_runtime_azure.go

```diff
@@ -0,0 +1,193 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+// DataFactoryIntegrationRuntimeAzure struct represents Azure Data Factory's
+// runtime flow.
+//
+// Resource information: https://azure.microsoft.com/en-us/services/data-factory/
+// Pricing information: https://azure.microsoft.com/en-us/pricing/details/data-factory/data-pipeline/
+type DataFactoryIntegrationRuntimeAzure struct {
+	Address string
+	Region  string
+
+	Cores       int64
+	ComputeType string
+
+	// "usage" args
+	MonthlyOrchestrationRuns *int64 `infracost_usage:"monthly_orchestration_runs"`
+}
+
+// DataFactoryIntegrationRuntimeAzureUsageSchema defines a list which represents the usage schema of DataFactoryIntegrationRuntimeAzure.
+var DataFactoryIntegrationRuntimeAzureUsageSchema = []*schema.UsageItem{
+	{Key: "monthly_orchestration_runs", DefaultValue: 0, ValueType: schema.Int64},
+}
+
+// PopulateUsage parses the u schema.UsageData into the DataFactoryIntegrationRuntimeAzure.
+// It uses the `infracost_usage` struct tags to populate data into the DataFactoryIntegrationRuntimeAzure.
+func (r *DataFactoryIntegrationRuntimeAzure) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid DataFactoryIntegrationRuntimeAzure struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *DataFactoryIntegrationRuntimeAzure) BuildResource() *schema.Resource {
+	runtimeFilter := "Cloud"
+
+	costComponents := []*schema.CostComponent{
+		r.computeCostComponent(),
+		dataFactoryOrchestrationCostComponent(r.Region, runtimeFilter, r.MonthlyOrchestrationRuns),
+		dataFactoryDataMovementCostComponent(r.Region, runtimeFilter),
+		dataFactoryPipelineCostComponent(r.Region, runtimeFilter),
+		dataFactoryExternalPipelineCostComponent(r.Region, runtimeFilter),
+	}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		UsageSchema:    DataFactoryIntegrationRuntimeAzureUsageSchema,
+		CostComponents: costComponents,
+	}
+}
+
+// computeCostComponent returns a cost component for cluster configuration.
+//
+// CPAPI contains 2 records with the same price, but one is newer and its
+// armSkuName is not empty thus using a non-empty filter.
+func (r *DataFactoryIntegrationRuntimeAzure) computeCostComponent() *schema.CostComponent {
+	productType := map[string]string{
+		"general":           "General Purpose",
+		"compute_optimized": "Compute Optimized",
+		"memory_optimized":  "Memory Optimized",
+	}[r.ComputeType]
+
+	return &schema.CostComponent{
+		Name:           fmt.Sprintf("Compute (%s, %d cores)", productType, r.Cores),
+		Unit:           "vCore-hours",
```

Comment:
Yeah, if we do `Compute (General Purpose, 272 vCores)` this might read better. Added it here for the same reason as with instances: seeing multiplied quantity without any explanation looked confusing. I'm going to change `cores` -> `vCores` 👍

Reply:
One other idea (and this seems like a bad idea 🙂), if "272 vCores" is the type in the name, maybe it would make sense to set the number of cores as the unit multipler so the unit can just be hours (I think the means the `price` field would be the hourly price for a 272 vCore instance)
```
Unit: "hours",
UnitMultiplier: r.Cores,
```

Reply:
Interesting, I think I like it :), this is how the pricing calculator shows it:
![image](https://user-images.githubusercontent.com/87082/163052539-a9a9418e-9694-4973-97bf-9e935869f19d.png)
But this may be confusing compared to vCore-hour on the pricing page 🤔.
![image](https://user-images.githubusercontent.com/87082/163052777-f1f4d0ce-2c77-4e86-9188-8392b5443b4f.png)

Reply:
I should have looked more into existing resources. Synapse workspace example:
```go
	var hourlyQuantity *decimal.Decimal
	if instances != nil && vCores != nil && hours != nil {
		hourlyQuantity = decimalPtr(vCores.Mul(*instances).Mul(*hours))
	}

	return &schema.CostComponent{
		Name:            name,
		Unit:            "vCore-hours",
		UnitMultiplier:  decimal.NewFromInt(1),
		MonthlyQuantity: hourlyQuantity,
```

The only difference is that `hours` come as usage based parameter, and we can keep 730h here as is.